openhcl/virt_mshv_vtl/src/lib.rs

```diff
@@ -1166,6 +1156,12 @@ pub trait ProtectIsolatedMemory: Send + Sync {
 
     /// Disables the overlay for the hypercall code page for a target VTL.
     fn disable_hypercall_overlay(&self, vtl: GuestVtl);
+
+    /// Alerts the memory protector that vtl 1 is ready to set protections on memory.
```

Comment:
suggestions on better names for enable_vtl1_protections and vtl1_protections_enabled or possibilities for restructuring? They just seem close enough to be confusing.

Reply:
So these aren't actually applying any protections themselves, they're just checking if the ability to apply protections is available/turning it on? Hmmmm. Maybe something like set_vtl1_protections_ready and vtl1_protections_ready? To make it clear that we're not actually protecting anything just yet?

Reply:
As far as I understand, it's not just applying, it also means enforcement, which is kinda interesting. Meaning that we'd actually go through with the vtl access check. But I think ready could still apply there.

I think we don't want to allow callers to set it to not ready. Do you think set* implies the ability to turn it off or set different values? Or do you think it's sufficient to say we don't provide any parameters, so that means it's a one-off.

Reply:
Yeah I think without parameters it becomes clear its a one off. Feel free to not take these if you don't like them though, its tricky i agree, but with good enough comments the current names may be alright.

Reply:
I like ready, but I might keep the "enabled" term because that's what used in the TLFS, so maybe that helps with drawing a line to what controls this. I'll switch to using set though and see if I can make the comments more helpful.